src/BotPlutusInterface/Effects.hs

```diff
@@ -106,6 +110,13 @@ data PABEffect (w :: Type) (r :: Type) where
   QueryChainIndex :: ChainIndexQuery -> PABEffect w ChainIndexResponse
   EstimateBudget :: TxFile -> PABEffect w (Either BudgetEstimationError TxBudget)
   SaveBudget :: Ledger.TxId -> TxBudget -> PABEffect w ()
+  SlotToPOSIXTime ::
```

Comment:
I started to get a feeling that instead of introducing new effects here for each cardano-api related query, we should have a more general `QueryNode` that uses this (this would also make later attempts to use cardano-api instead of the cli or chain-index a bit simpler)

Reply:
Agreed yes, let's make that a new issue (the problem with refactoring issues is that they tend to be forgotten, staying in the backlog forever, so I like to couple them with some useful feature)

Reply:
Made an issue #109